### DISCUSSION TITLE
某些应用界面内规则不执行，需要手动触发

### DISCUSSION THREAD

**Discussion by abuc1900** [gkd-kit/gkd#825] [Q&A] [answered]
呼叫大佬！发现GKD一个BUG，在深黑色背景和黑色状态栏下GKD不能很快自动识别点击控件，只有我下拉状态栏屏幕上部分变白色它才自动识别点击，请问如何破解？
https://i.gkd.li/i/18155332
![Screenshot_2024-12-17-14-19-10-25_521b94d5548222ec15c65476e6d19023](https://github.com/user-attachments/assets/997316e0-2125-487e-9dbe-aaaf548e1cfe)

**Comment by abuc1900** [comment]
@lisonge 

设置了matchRoot: true,无效
{
  name: '[ChangeMe]规则名称-2024-12-17 02:26:26',
  key: 1,
  desc: '[ChangeMe]本规则由GKD网页端审查工具生成',
  actionMaximum: 1,
  rules: [
    {
      matches: [
        'Button[desc="确定"][clickable=true][visibleToUser=true]'
      ],
      fastQuery: true,
      matchRoot: true,
      snapshotUrls: [
        'https://i.gkd.li/i/18155332'
      ],
      exampleUrls: [
        'https://e.gkd.li/undefined'
      ],
      activityIds: [
        'com.lxy.qr.app.MainActivity'
      ]
    }
  ]
}

**Comment by lisonge** [comment]
试试 https://gkd.li/api/interfaces/RawAppGroup#forcedtime

**Reply by abuc1900** [reply]
> 试试 https://gkd.li/api/interfaces/RawAppGroup#forcedtime

大大，这个有点复杂我是小白，只懂简单的模仿使用选择器，术语看了头大，状态栏黑色或背景黑色不能自动识别，白色状态栏和白色背景app GKD识别就会很快😭

**Reply by lisonge** [reply, marked as answer]
```json5
{
  name: '[ChangeMe]规则名称-2024-12-17 02:26:26',
  key: 1,
  desc: '[ChangeMe]本规则由GKD网页端审查工具生成',
  actionMaximum: 1,
  rules: [
    {
      matches: ['Button[desc="确定"][clickable=true][visibleToUser=true]'],
      fastQuery: true,
      matchRoot: true,
      snapshotUrls: ['https://i.gkd.li/i/18155332'],
      exampleUrls: ['https://e.gkd.li/undefined'],
      activityIds: ['com.lxy.qr.app.MainActivity'],
      forcedTime: 10000,
    },
  ],
}
```

**Reply by abuc1900** [reply]
> ```js
> {
>   name: '[ChangeMe]规则名称-2024-12-17 02:26:26',
>   key: 1,
>   desc: '[ChangeMe]本规则由GKD网页端审查工具生成',
>   actionMaximum: 1,
>   rules: [
>     {
>       matches: ['Button[desc="确定"][clickable=true][visibleToUser=true]'],
>       fastQuery: true,
>       matchRoot: true,
>       snapshotUrls: ['https://i.gkd.li/i/18155332'],
>       exampleUrls: ['https://e.gkd.li/undefined'],
>       activityIds: ['com.lxy.qr.app.MainActivity'],
>       forcedTime: 10000,
>     },
>   ],
> }
> ```

有点改善，但还是越黑反应越慢，无解了😬

**Reply by lisonge** [reply]
麻烦把 apk 发出来看看

**Reply by abuc1900** [reply]
> 麻烦把 apk 发出来看看

大大，打开【搜剧】APP(https://souju.app) 搜【家族计划 】
进入APP后点击这部影片播放，这部背景比较黑GKD就不反应了

**Reply by abuc1900** [reply]
> 麻烦把 apk 发出来看看

APP文件大于25M发不上去

**Reply by lisonge** [reply]
https://www.123pan.com/

**Reply by abuc1900** [reply]
> https://www.123pan.com/

https://www.123684.com/s/bsl3jv-sOrgA

**Reply by abuc1900** [reply]
播放我是刑警反应也慢，要手动一下，如果是白色状态栏GKD都不需要等待就能直接自动识别点击了

**Reply by lisonge** [reply]
无法正常打开使用

安装后打开直接跳转浏览器 zbk.zpp 链接然后退出了，下载这个网页的 app 也是一样

使用梯子打开也是跳转外部网页

**Reply by abuc1900** [reply]
> 无法正常打开使用
> 
> 安装后打开直接跳转浏览器 zbk.zpp 链接然后退出了，下载这个网页的 app 也是一样
> 
> 使用梯子打开也是跳转外部网页

奇怪，我这里能直接打开这个123链接，不行算了不搞了，感谢大佬的回复！
![Screenshot_2024-12-17-16-18-49-02_df198e732186825c8df26e3c5a10d7cd](https://github.com/user-attachments/assets/1fad30bf-8c85-4cae-96ca-e3ab146b5d97)

**Reply by lisonge** [reply]
你这个问题应该是获取节点耗时过长导致的，你把 matchRoot 去掉看看行不

**Reply by lisonge** [reply]
> 奇怪，我这里能直接打开这个123链接

我能下载安装，但是安装后在桌面打开出现我上面描述的行为，导致我无法使用这个应用

**Comment by abuc1900** [comment]
要把梯子关了，再进入APP



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年12月17日(周二) 下午4:24
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [gkd-kit/gkd] 某些应用界面内规则不执行，需要手动触发 (Discussion #825)




  
奇怪，我这里能直接打开这个123链接
  
我能下载安装，但是安装后在桌面打开出现我上面描述的行为，导致我无法使用这个应用

**Reply by lisonge** [reply]
- 打开梯子-打开应用-提示需关闭梯子后打开-被跳转浏览器
- 关闭梯子-打开应用-被跳转浏览器

总之无论我怎样都是跳转外部浏览器，不管我是 wifi 还在 流量，是否打开梯子

**Comment by abuc1900** [comment]
算了，不搞他了，反应慢点就反应慢点



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年12月17日(周二) 下午4:30
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [gkd-kit/gkd] 某些应用界面内规则不执行，需要手动触发 (Discussion #825)




  
打开梯子-打开应用-提示需关闭梯子后打开-被跳转浏览器
 
关闭梯子-打开应用-被跳转浏览器
  
总之无论我怎样都是跳转外部浏览器，不管我是 wifi 还在 流量，是否打开梯子
